### DISCUSSION TITLE
Include starter dependencies configuration properties

### DISCUSSION THREAD

**Discussion by roded** [rodnansol/spring-configuration-property-documenter#69] [Ideas]
Hi, 
It would be great if this library would be able to include the configuration properties for dependent starters in the produced documentation.

For example, I have project-a which uses starter-a Spring Boot starter module which itself defines several properties using `@ConfigurationProperties`. The starter properties can be regarded as properties of project-a. 

Thanks

**Comment by nandorholozsnyak** [comment]
Hello @roded 

If I understood correctly, you should be already able to do it, please check the following resources:

- https://github.com/rodnansol/spring-configuration-property-documenter/discussions/30#discussioncomment-5183100
- https://github.com/rodnansol/spring-configuration-property-documenter/blob/master/samples/multi-module/multi-module-docs/pom.xml#L255

Unfortunately the samples are not "documented", but you can check them out in the **samples** folder.

**Reply by roded** [reply]
Hi @nandorholozsnyak , 
Got it, so I should explicitly define which modules should be included in the configuration documentation. That works well.
I do wonder though whether it's possible to derive the required modules from the Gradle dependencies. FYC.

Thanks!

**Reply by nandorholozsnyak** [reply]
Sounds fun, but I think both build tools' API is able to provide this list at runtime. If you think it should be a feature, please open a pull request.
The problem I would foresee is that starters are having a tons of properties, you might want to limit them to the used ones, that would require another scan in the project to discover the used properties, a more complex stuff, but nothing is impossible :D

**Reply by roded** [reply]
Oh, I have not considered getting the project's dependencies at Gradle task runtime and adding `AggregationInput`s dynamically. I'll look into how to do that. 

Thanks again @nandorholozsnyak .
